### PR TITLE
chore: add error msg in the logs when gw req fails

### PR DESCRIPTION
**Pull request by mihir20**
# Description

add error msg in the logs when gw req fails

## Linear Ticket

https://linear.app/rudderstack/issue/PIPE-1797/better-error-logging-for-ingestion-requests

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: gateway/handle.go [hunk 1]
```diff
@@ -663,7 +663,7 @@ func (gw *Handle) internalBatchHandlerFunc() http.HandlerFunc {
 			stat.Report(gw.stats)
 			goto requestError
```
Comment (BonapartePC) [inline gateway/handle.go:664]: All the errors going to `requestError` should be printing error message. Examples: Line 664 and 680

File: gateway/handle.go [hunk 2]
```diff
@@ -706,14 +706,17 @@ func (gw *Handle) internalBatchHandlerFunc() http.HandlerFunc {
 		return
 
 	requestError:
-		errorMessage = err.Error()
+		if errorMessage == "" {
+			errorMessage = err.Error()
+		}
 		status = response.GetErrorStatusCode(errorMessage)
 		responseBody = response.GetStatus(errorMessage)
-		gw.logger.Infon("response",
```
Comment (BonapartePC) [inline gateway/handle.go:712]: Let's also log properties if they are available after unmarshalling

**Review by BonapartePC — APPROVED** [review]

File: gateway/handle.go [hunk 3]
```diff
@@ -763,14 +766,14 @@ func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byt
 	if err != nil {
 		stat.RequestFailed(response.InvalidJSON)
 		stat.Report(gw.stats)
-		return nil, errors.New(response.InvalidJSON)
+		return nil, response.InvalidJSON, fmt.Errorf("unmarshalling request body: %w", err)
```
Comment (lvrach) [inline gateway/handle.go:769]: This is not an elegant way to do this. Using wrapping and using `if errors.As(err, &e)` to extract response error would be a better approach. https://go.dev/blog/go1.13-errors

If this is too complex given the time constrains, I will log the error here.

**Review by fracasula — CHANGES_REQUESTED** [review]

File: gateway/handle.go [hunk 4]
```diff
@@ -763,13 +764,17 @@ func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byt
...
+			logger.NewStringField("body", string(body)),
+			logger.NewErrorField(err))
 		return nil, errors.New(response.InvalidJSON)
 	}
 	gw.requestSizeStat.Observe(float64(len(body)))
 
 	if len(messages) == 0 {
 		stat.RequestFailed(response.NotRudderEvent)
 		stat.Report(gw.stats)
+		gw.logger.Errorn("no messages in request", logger.NewStringField("body", string(body)))
```
Comment (fracasula) [inline gateway/handle.go:777]: Again, let's not dump the body. Payloads should be encrypted and uploaded somewhere safe, not logged like this.

Comment (fracasula) [reply gateway/handle.go:777]: Same applies to all other places where the message payload is being dumped.

Comment (fracasula) [reply gateway/handle.go:777]: Uploading should be encrypted imo.

**Review by fracasula — APPROVED** [review]

**Pull request merged**